### PR TITLE
Make Python RPC handler does not hold module in global variable

### PR DESCRIPTION
**Pull request by xush6528**
Summary:
# Problem

ProcessGroupAgent used in test_rpc has SIGSEGV on exiting.

# Solution

It was because the python module was unpexceted destructed twice.

Test Plan: `

Differential Revision: D17127093

### CODE REVIEW

File: torch/csrc/distributed/rpc/python_rpc_handler.cpp
Comment (mrshenli) [inline torch/csrc/distributed/rpc/python_rpc_handler.cpp]: When this can be none? And why import it again would work?

**Review by mrshenli — COMMENTED** [review]
Could you please elaborate more on why `py::object module_` was destructed twice? 

BTW @zhaojuanmao is working on some related changes that make RPC handler a singleton. Please get a review from her as well.

**Comment by zhaojuanmao** [comment]
@xush6528 I have two diffs stacked to fix the tests and codes, do you want to wait for that? I hold to  export it to OSS now because current Multi processing test case class is still broken.  it is better to land our diffs after the test lib is fixed cleanly.

**Review by zhaojuanmao — APPROVED** [review]
approving it as tmp fix to unblock, #25656 and its stack need more time to debug

please make sure CI tests are green after rebasing

File: torch/csrc/distributed/rpc/python_rpc_handler.cpp
Comment (zhaojuanmao) [reply torch/csrc/distributed/rpc/python_rpc_handler.cpp]: also address the comment

Comment (xush6528) [reply torch/csrc/distributed/rpc/python_rpc_handler.cpp]: Will remove this. import failure means wrong code.

**Pull request closed without merging**
